ContextDesktop/src/main/java/com/github/migi_1/Context/enemy/EnemyFactory.java

```diff
@@ -1,20 +1,76 @@
 package com.github.migi_1.Context.enemy;
 
-import com.github.migi_1.Context.obstacle.AbstractObstacleFactory;
-import com.github.migi_1.Context.obstacle.Obstacle;
+import java.util.ArrayList;
 
+import com.github.migi_1.Context.model.entity.Carrier;
+import com.jme3.bounding.BoundingBox;
+import com.jme3.math.Vector3f;
 
 /**
  * This class is a factory that produces Enemy objects.
  * @author Damian
  *
  */
-public class EnemyFactory extends AbstractObstacleFactory {
-
-    @Override
-    public Obstacle produce() {
-        // TODO Auto-generated method stub
-        return null;
+public class EnemyFactory {
+    
+    private BoundingBox levelPieceBoundingBox;
+    private float levelPieceLength;
+    private float levelPieceWidth;
+    private ArrayList<Carrier> carriers;
+    
+    /**
+     * Constructor of the EnemyFactory.
+     * @param levelPieceBoundingBox the boundingbox of the levelPiece.
+     * @param levelPieceLength Length of the levelPiece.
+     * @param levelPieceWidth Width of the levelPiece.
+     * @param carriers the carriers on which the spawning will be based.
+     */
+    public EnemyFactory(BoundingBox levelPieceBoundingBox, float levelPieceLength,
+            float levelPieceWidth, ArrayList<Carrier> carriers) {
+        this.levelPieceBoundingBox = levelPieceBoundingBox;
```

Comment:
the enemy spawner also needs information of the levelPiece so it can give this information directly, it wouldn't be any cleaner and I feel it would require changing more things than it is worth